plugin/src/main/groovy/com/novoda/staticanalysis/internal/lint/LintConfigurator.groovy

```diff
@@ -0,0 +1,64 @@
+package com.novoda.staticanalysis.internal.lint
+
+import com.novoda.staticanalysis.StaticAnalysisExtension
+import com.novoda.staticanalysis.internal.Configurator
+import com.novoda.staticanalysis.internal.Violations
+import org.gradle.api.NamedDomainObjectContainer
+import org.gradle.api.Project
+import org.gradle.api.Task
+
+class LintConfigurator implements Configurator {
+
+    private final Project project
+    private final Violations violations
+    private final Task evaluateViolations
+
+    static LintConfigurator create(Project project,
+                                   NamedDomainObjectContainer<Violations> violationsContainer,
+                                   Task evaluateViolations) {
+        Violations violations = violationsContainer.maybeCreate('Lint')
```

Comment:
I'm confused by this `maybeCreate('Lint')`
This means that the container is supposed to have a named closure called `Lint` but I couldn't find another usage. Not in the tests for example.

Reply:
I am also confused since I don't fully understand what's going in here. Tbh I just copy pasted it from the other configurators. From what I understand we pass the same `Violations` to each `Configurator`. But I don't understand how this relates to it.

Reply:
Maybe @mr-archano can enlighten us around this.

Reply:
We just hold all the violations in the plugin extension to keep the output of each tool output in memory before the final evaluation (see [`StaticAnalysisExtension.allViolations`](https://github.com/novoda/gradle-static-analysis-plugin/blob/develop/plugin/src/main/groovy/com/novoda/staticanalysis/StaticAnalysisExtension.groovy#L72)).
Each tool is now responsible of creating the `Violations` entry in the container, that is then passed to the [`evaluateViolations`](https://github.com/novoda/gradle-static-analysis-plugin/blob/develop/plugin/src/main/groovy/com/novoda/staticanalysis/StaticAnalysisPlugin.groovy#L29) task.

As already mentioned IRL couple of times this is a pretty lame way of collecting all the info, and I already logged an issue to get rid of this in the next future: https://github.com/novoda/gradle-static-analysis-plugin/issues/43